c/experimental/stf/include/cccl/c/experimental/stf/stf.h

```diff
@@ -0,0 +1,224 @@
+#include <assert.h>
```

Comment:
Might be better to not include this header if possible. I know you define some functions here that use assert, but could you move the function definitions to a source file instead?